packages/glimmer-runtime/lib/builder.ts

```diff
@@ -89,13 +91,19 @@ class GroupedElementOperations implements ElementOperations {
   }
 
   addAttribute(name: InternedString, reference: PathReference<string>) {
+    if (name === 'style') {
+      this.env.warn(`${name}-attribute`);
+    }
     if (this.element.tagName === 'INPUT' && name === 'value') {
       this.group.push(new EmptyStringResetingProperty(this.element, name, reference));
     }
     this.group.push(new NonNamespacedAttribute(this.element, name, reference));
   }
 
   addAttributeNS(namespace: InternedString, name: InternedString, reference: PathReference<string>) {
+    if (name === 'style') {
+      this.env.warn(`${name}-attribute`);
```

Comment:
I want some type of API here that allows us to disambiguate on the calling side or I can just special case the method and call it a day.